impress-org/givewp
# New: Migrate legacy give-recurring subscriptions into core

Resolves #6166
🛑  Depends on [Give Recurring PR](https://github.com/impress-org/give-recurring/pull/1111)

## Description



This is the first step into identifying _Subscriptions_ as a core concept. We are relocating all the key subscription classes into give core.   All that should be left in give-recurring is logic that uses these concepts and/or extends give core - not WordPress.

## Affects



- A new `LegacySubscriptions` domain & service provider with the legacy files for Subscriptions and related database functionality.
- added `@property-read Give_Recurring_DB_Subscription_Meta $subscription_meta` to main Give class

The classes below have been relocated to Give Core from give-recurring
- `Give_Recurring_Cache`
- `Give_Recurring_Cron`
- `Give_Recurring_DB_Subscription_Meta`
- `Give_Recurring_Subscriber`
- `Give_Subscription`
- `Give_Subscriptions_API`
- `Give_Subscriptions_DB`

## Testing Instructions



- Make sure you're recurring branch is aligned with this PR https://github.com/impress-org/give-recurring/pull/1111
- Create a donation form with recurring
- Donate with recurring
- Uninstall recurring and donate with the same form

## Pre-review Checklist
